src/shadow-embed.js

```diff
@@ -14,11 +14,20 @@
  * limitations under the License.
  */
 
+import {ShadowCSS} from '../third_party/webcomponentsjs/ShadowCSS';
 import {ampdocFor} from './ampdoc';
+import {dev} from './log';
 import {extensionsFor} from './extensions';
 import {copyRuntimeStylesToShadowRoot} from './style-installer';
 
 
+/** @const {!RegExp} */
+const CSS_SELECTOR_BEG_REGEX = /[^\.\-\_0-9a-zA-Z]/;
+
+/** @const {!RegExp} */
+const CSS_SELECTOR_END_REGEX = /[^\-\_0-9a-zA-Z]/;
```

Comment:
Not quite. There are actually never going to be `{}` in the selectors since they are already tokenized by the browser. But we need to disambiguate specific naming issues. E.g. 'body' and `body.x` needs to be renamed, while `.body`, `x-body`, `body-x` and many others do not.